### PR TITLE
build(docker): upgrade base image to ubuntu 24.04

### PR DESCRIPTION
**Pull request by wirespecter**
Closes https://github.com/reanahub/reana/issues/808
Old PR: https://github.com/reanahub/reana-workflow-engine-yadage/pull/270

### CODE REVIEW

File: setup.py
Comment (mdonadoni) [inline setup.py:1]: You should also update the classfiers from `Python :: 3.8` to `Python :: 3.12`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -156,7 +156,7 @@ jobs:
 
       - name: Install Python dependencies
         run: |
-          pip install --upgrade pip 'setuptools<56' py
+          pip install --upgrade pip 'setuptools<71' py
```
Comment (mdonadoni) [inline .github/workflows/ci.yml:159]: Ditto

File: Dockerfile [hunk 2]
```diff
@@ -32,12 +30,12 @@ RUN apt-get update -y && \
...
       openssl \
-      python3.8 \
-      python3.8-dev \
+      python3.12 \
+      python3.12-dev \
       python3-pip \
       unzip \
       vim-tiny && \
-    pip install --no-cache-dir --upgrade pip 'setuptools<71' && \
+    pip install --no-cache-dir --upgrade 'setuptools<71' && \
```
Comment (mdonadoni) [inline Dockerfile:38]:
```suggestion
    pip install --no-cache-dir --upgrade setuptools && \
```
With Ubuntu 24.04 this should not be needed anymore, could you please double check also in the other PRs?

File: setup.py [hunk 3]
```diff
@@ -34,9 +34,7 @@
     ],
     # Using older jq on amd64 due to https://github.com/reanahub/reana-demo-bsm-search/issues/21
```
Comment (mdonadoni) [inline setup.py:35]:
```suggestion
```
This line can be removed, as now we are using the same new version of jq for both amd64 and arm64

**Review by tiborsimko — APPROVED** [review]

**Pull request merged**